### PR TITLE
chore: cleanup a few lint errors

### PR DESCRIPTION
**Pull request by maxkorp**
## Description

Fixes some `any` uses and some deprecations. 

uses of `any` should be pretty self explanatory.

There are also a few deprecated method calls (`substr` vs `substring` and `Assert.equals` vs `Assert.strictEquals`. I checked files manually for deprecations in vscode. I tried adding a lint rule (https://github.com/gund/eslint-plugin-deprecation) but it doesn't seem to catch everything (didn't catch the substr for example). Added a pretty big install change so I left it off. 

## Performance impact

None

## Security impact

None. The only change to the JS output is `assert.strictEqual` vs `assert.equal` in `current.js`

## Checklist

- [x] My code matches the project's code style and `yarn lint:fix` passes.
- [x] I've added tests for the new feature, and `yarn test` passes.
- [x] I have detailed the new feature in the relevant documentation.
- [x] I have added this feature to 'Pending' in the `RELEASE_NOTES.md` file (if one exists).
- [x] If this is a breaking change I've explained why.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  db:
+    image: maxkorp/postgis:11.4-2.5.2
+    container_name: migrate-db
+    restart: on-failure
+    env_file:
+      - .env
+    expose:
+      - 5432
+    ports:
+      - 5432:5432
```
Comment (benjie) [inline docker-compose.yml:12]: Please remove this :+1:

File: src/commands/run.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ interface RunArgv extends CommonArgv {
   rootDatabase?: boolean;
 }
 
-export async function run(
+export async function run<T = void>(
```
Comment (benjie) [inline src/commands/run.ts:21]: `void` is definitely wrong here, `never` would be a more appropriate type but I think `<T = any>` is preferred.
```suggestion
export async function run<T = any>(
```

File: src/instrumentation.ts [hunk 3]
```diff
@@ -4,11 +4,18 @@ import indent from "./indent";
 import { Client } from "./pg";
 import { ParsedSettings } from "./settings";
 
-export async function runQueryWithErrorInstrumentation(
+interface InstrumentationError extends Error {
+  severity: string;
+  code: string;
+  detail: string;
+  hint: string;
```
Comment (benjie) [inline src/instrumentation.ts:11]: I don't think we can rely on any of these existing; also there's the `_gmLogged` hack we do:

```suggestion
  severity?: string;
  code?: string;
  detail?: string;
  hint?: string;
  _gmlogged?: boolean;
  _gmMessageOverride?: string;
```

File: src/instrumentation.ts [hunk 4]
```diff
@@ -4,11 +4,18 @@ import indent from "./indent";
...
+}
+
+export async function runQueryWithErrorInstrumentation<T = void>(
```
Comment (benjie) [inline src/instrumentation.ts:14]:
```suggestion
export async function runQueryWithErrorInstrumentation<T = any>(
```

File: src/instrumentation.ts (hunk 3 above)
Comment (benjie) [reply src/instrumentation.ts:11]: That `gmlogged` doesn't have a capital L pains me.

File: src/instrumentation.ts [hunk 5]
```diff
@@ -69,7 +76,7 @@ export const logDbError = ({ logger }: ParsedSettings, e: Error): void => {
       chalk.red.bold(`🛑 Error occurred whilst processing migration`),
     );
   }
-  const { severity, code, detail, hint } = e as any;
+  const { severity, code, detail, hint } = e as InstrumentationError;
```
Comment (benjie) [inline src/instrumentation.ts:79]: Change the function declaration to:

```
export const logDbError = ({ logger }: ParsedSettings, error: Error): void => {
  const e = error as InstrumentationError;
  e._gmlogged = true;
```

Then we can:

```suggestion
  const { severity, code, detail, hint } = e;
```

**Review by benjie — CHANGES_REQUESTED** [review]
Looks good; thanks for not adding any dependencies!

File: docker-compose.yml (hunk 1 above)
Comment (maxkorp) [reply docker-compose.yml:12]: Shoot, I absolutely did not mean to include this file at all

File: src/instrumentation.ts (hunk 3 above)
Comment (maxkorp) [reply src/instrumentation.ts:11]: is that something from inside migrate? can we change that?

File: src/commands/run.ts (hunk 2 above)
Comment (maxkorp) [reply src/commands/run.ts:21]: So, I got this by following that we return the call to pgReal.withClient, which is `<T = void>` and returns a `Promise<T>`? Happy to make it an any (although es-lint will then complain) if you're sure. To get around the complaining, can I just eslint-ignore that line still? is that still even a thing?

File: src/instrumentation.ts (hunk 4 above)
Comment (maxkorp) [reply src/instrumentation.ts:14]: Ah yeah ok I see. I'll eslint-ignore that then, yeah?

**Comment by maxkorp** [comment]
Ok pushed up anything. Since the resolved convos get hidden I believe, I'll relay them here

1. Is `_gmlogged` a migrate thing? Can we (and do we) want to change it to `_gmLogged`?

2. In `run` in run.ts, we use `withQuery` (from `pgReal`), and pass in `runQueryWithErrorInstrumentation`. Is there something wrong that the T type from `runQueryWithErrorInstrumentation` doesn't get bubbled up to `withQuery` and there by `run`? Or is that expected in typescript? 

Thanks for the help here. My typescript fu is super weak; since I'm used to either wild west vanilla JS or extremely strict/rigid ocaml, I don't really know where to lay my expectations of behaviors with typescripts in-betweeny level of strictness. Just trying to clean up the assault I get when running tests etc.

File: src/instrumentation.ts (hunk 3 above)
Comment (benjie) [reply src/instrumentation.ts]: Yeah we could change it, I'm not sure what the effects would be...

**Review by benjie — CHANGES_REQUESTED** [review]
Change line 56 of `run.ts` to 

```ts
    runQueryWithErrorInstrumentation<T>(pgClient, sql, filename),
```

Other than that I think we're good to go. Let's leave `_gmlogged` -> `_gmLogged` for a separate PR :wink:

**Comment by benjie** [comment]
(I also commented in the collapsed discussions.)

**Review by benjie — APPROVED** [review]
Looks good to me - thanks! :raised_hands:

**Pull request merged**